### PR TITLE
drop julia v0.5 support

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

File: src/Quadrature/quadrature.jl [hunk 1]
```diff
@@ -100,10 +100,10 @@ julia> getpoints(qr)
 """
 getpoints(qr::QuadratureRule) = qr.points
 
-(::Type{QuadratureRule{dim, shape}}){dim, shape}(order::Int) = QuadratureRule{dim, shape}(:legendre, order)
+QuadratureRule{dim, shape}(order::Int) where {dim, shape} = QuadratureRule{dim, shape}(:legendre, order)
 
 # Special case for face integration of 1D problems
-function (::Type{QuadratureRule{0, RefCube}})(quad_type::Symbol, order::Int)
+function QuadratureRule{0, RefCube}(quad_type::Symbol, order::Int)
```
Comment (fredrikekre) [inline src/Quadrature/quadrature.jl]: Hmm... How to do this with the 0?

Comment (fredrikekre) [inline src/Quadrature/quadrature.jl]: Isn't this exactly what we do here? https://github.com/KristofferC/Tensors.jl/blob/b72cad701c5007a417a882f939dfe6bb2616f864/src/Tensors.jl#L96

**Comment by fredrikekre** [comment]
Plz review when you have some time.

File: src/Quadrature/quadrature.jl (hunk 1 above)
Comment (KristofferC) [inline src/Quadrature/quadrature.jl]: Everything works except this?

Comment (fredrikekre) [inline src/Quadrature/quadrature.jl]: Yes, I think so. Can't even `using JuAFEM`:
```
julia> include("src/JuAFEM.jl")
ERROR: LoadError: LoadError: syntax: invalid type parameter name "0"
Stacktrace:
 [1] include_from_node1(::String) at ./loading.jl:569
 [2] include(::String) at ./sysimg.jl:14
 [3] include_from_node1(::String) at ./loading.jl:569
 [4] include(::String) at ./sysimg.jl:14
while loading /home/fredrik/Dropbox/Programming/JuAFEM.jl/src/Quadrature/quadrature.jl, in expression starting on line 106
while loading /home/fredrik/Dropbox/Programming/JuAFEM.jl/src/JuAFEM.jl, in expression starting on line 36
```

Comment (KristofferC) [inline src/Quadrature/quadrature.jl]: Is there some deprecation warning for this?

Comment (fredrikekre) [inline src/Quadrature/quadrature.jl]: What do you mean? It is not a problem in `Tensors`.

Comment (fredrikekre) [inline src/Quadrature/quadrature.jl]: Oh, you mean if this change is necessary, no I guess not.

**Pull request merged**